docs/docsite/rst/vmware/scenario_clone_template.rst

```diff
@@ -0,0 +1,124 @@
+          name: testvm_2
+          template: template_el7
+          datacenter: dc1
+          folder: /dc1/vm
+          state: poweredon
+          wait_for_ip_address: yes
+
+
+Since Ansible utilizes the VMware API to perform actions, in this use case we will be connecting directly to the API from our localhost. This means that our playbooks will not be running from the vCenter or ESXi Server. We do not necessarily need to collect facts about our localhost, so the *gather_facts* parameter will be disabled. You can run these modules against another server that would then connect to the API if your localhost does not have access to vCenter. If so, the required Python modules will need to be installed on that target server.
+
+To begin, there are a few bits of information we will need. First and foremost is the hostname of the ESXi server or vCenter server. After this, we will need the username and password for this server. For now, you will be entering these directly, but in a more advanced playbook this can be abstracted out and stored in a more secure fashion [1][2]. If your vCenter or ESXi server is not setup with proper CA certificates that can be verified from the Ansible server, then it is necessary to disable validation of these certificates by using the *validate_certs* parameter. To do this you need to set `validate_certs=False` in your playbook.
+
+Now we get into supplying the information about the VM we will be creating. We will need to give this VM a name. It must conform to all VMware requirements for naming conventions.  Next we will need the display name of the template that we will be cloning. This must match exactly with what is displayed in VMware.  A folder to place this virtual machine in can then be specified. This can either be a relative path or a full path to the folder including the Datacenter. We must then specify a state for the VM.  This simply tells it which action we want to take, in this case we will be ensure that the VM exists and is powered on.  An optional parameter is *wait_for_ip_address*, this will tell Ansible to wait for the machine to fully boot up and VMware Tools is running before completing this task.
+
+
+What to expect
+--------------
+
+- You will see a bit of JSON output after this playbook completes. This output shows various parameters that are returned from the module and from vCenter about the newly created VM.
+
+- State is changed to *True* which notifies that the virtual machine is built using given template. The module will not complete until the clone task in VMware is finished. This can take some time depending on your environment.
+
+- If you utilize the *wait_for_ip_address* parameter, then it will also increase the clone time as it will wait until virtual machine boots into the OS and an IP Address has been assigned to the given NIC.
+
+
+
+Troubleshooting
+---------------
+
+Things to inspect
+
+- Check if the values provided for username and password are correct
```

Comment:
This section would be stronger if we added the error messages the user would see for each problem - for example (I'm making up the error message here, because I don't know what it is), "If your playbook fails with `datacenter not found`, make sure the datacenter you provided is available." For even stronger docs, add a short note on how to perform each troubleshooting step (again, I'm totally making up this example): " . . . make sure the datacenter you provided is available by typing `vmware datacenter show`." And for the third level, add example output of the troubleshooting step.

Reply:
I will add this in separate page with separate PR.